app/crypto/EncryptedValue.scala

```diff
@@ -38,7 +38,7 @@ class AesGCMCrypto @Inject() () {
 
   val IV_SIZE                       = 96
   val TAG_BIT_LENGTH                = 128
-  val ALGORITHM_TO_TRANSFORM_STRING = "AES/GCM/PKCS5Padding"
+  val ALGORITHM_TO_TRANSFORM_STRING = "AES/GCM/NoPadding"
```

Comment:
Are these changes supposed to be here?

Reply:
Yes, Java 21 will not allow PKCS5Padding to be used with GCM.
I confirmed this with Platops: https://hmrcdigital.slack.com/archives/C0GS60DK2/p1733923750290349